test/functional/provider/ruby_spec.lua

```diff
@@ -0,0 +1,96 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local eq = helpers.eq
+local feed = helpers.feed
+local clear = helpers.clear
+local funcs = helpers.funcs
+local meths = helpers.meths
+local insert = helpers.insert
+local expect = helpers.expect
+local command = helpers.command
+local write_file = helpers.write_file
+local curbufmeths = helpers.curbufmeths
+
+do
+  clear()
+  command('let g:prog = provider#ruby#Detect()')
+  local prog = meths.get_var('prog')
+
+  if prog == '' then
+    pending(
+    "Couldn't find the neovim RubyGem. Install it with `gem install neovim`.",
+    function() end)
+    return
+  end
+end
+
+before_each(function()
+  clear()
+end)
+
+describe('ruby feature test', function()
+  it('works', function()
+    eq(1, funcs.has('ruby'))
+  end)
+end)
+
+describe(':ruby command', function()
+  it('evaluates ruby', function()
+    command('ruby VIM.command("let g:set_by_ruby = [100, 0]")')
+    eq({100, 0}, meths.get_var('set_by_ruby'))
+  end)
+
+  it('supports nesting', function()
+    command([[ruby VIM.command('ruby VIM.command("let set_by_nested_ruby = 555")')]])
+    eq(555, meths.get_var('set_by_nested_ruby'))
+  end)
+end)
+
+describe(':rubyfile command', function()
+  it('evaluates a ruby file', function()
+    local fname = 'rubyfile.rb'
+    write_file(fname, 'VIM.command("let set_by_rubyfile = 123")')
+    command('rubyfile rubyfile.rb')
+    eq(123, meths.get_var('set_by_rubyfile'))
+    os.remove(fname)
```

Comment:
I suppose it's possible? It's just doing a `File.read`, which should close the file immediately. I actually changed it on master to just `load` the file instead of read + eval, so it's possible that will resolve it. I'll give it a try later today.